src/scripts/main.js

```diff
@@ -2,10 +2,23 @@
 
 function waitFor(element, eventName) {
   // write your code here
+  return new Promise((resolve) => {
+    element.addEventListener(eventName, function handler() {
+      const message = `It was ${eventName} on the element: ${element.nodeName}, id: ${element.id}.`;
+
+      resolve(message);
+      element.removeEventListener(eventName, handler);
+    });
+  });
 }
 
 const printMessage = (message) => {
   // write your code here
+  const messageDiv = document.createElement('div');
+
+  messageDiv.classList.add('message');
+  messageDiv.textContent = message;
+  document.body.appendChild(messageDiv);
 };
```

Comment:
In the `printMessage` function, you are creating a new DOM element every time the function is called. This could potentially lead to performance issues if the function is called a large number of times. Consider reusing the same DOM element and just updating its text content instead.